include/zephyr/linker/linker-defs.h

```diff
@@ -294,17 +294,25 @@ extern char z_kobject_data_begin[];
 #ifdef CONFIG_THREAD_LOCAL_STORAGE
 extern char __tdata_start[];
 extern char __tdata_end[];
-extern char __tdata_size[];
 extern char __tdata_align[];
 extern char __tbss_start[];
 extern char __tbss_end[];
-extern char __tbss_size[];
 extern char __tbss_align[];
 extern char __tls_start[];
 extern char __tls_end[];
 extern char __tls_size[];
+
+#ifdef CONFIG_RISCV_CMODEL_MEDANY
```

Comment:
although this does address the exact issue reported, then I think we would want a solution which covers this more broadly.

Also because there are other toolchains where tricks must be used to achieve similar, and where runtime calculation would simplify things.